libs/constrained-generators/src/Constrained/Examples/Map.hs

```diff
@@ -85,32 +85,36 @@ mapRestrictedValues :: Specification BaseFn (Map (Either Int ()) Int)
 mapRestrictedValues = constrained $ \m ->
   [ assert $ sizeOf_ (dom_ m) ==. 6
   , forAll' m $ \k v ->
-      caseOn
-        k
-        (branch $ \_ -> 20 <=. v)
-        (branch $ \_ -> True)
+      [ caseOn
+          k
+          (branch $ \_ -> 20 <=. v)
+          (branch $ \_ -> True)
+      , v `dependsOn` k
+      ]
   ]
```

Comment:
This PR is trying to fix the 'v' before 'k' behavior. So is this depends on forcing that behavior, or demonstrsting that after the fix, that such a depends on actuall can have an effect.  Pleas add a comment to this test case.